src/app/services/auth.service.ts

```diff
@@ -1,76 +1,74 @@
 import { Injectable } from '@angular/core';
 import { AngularFireAuth } from '@angular/fire/compat/auth';
-import { BehaviorSubject, catchError, from, Observable, of, throwError } from 'rxjs';
+import { BehaviorSubject, Observable, throwError, from } from 'rxjs';
+import { catchError } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-  constructor(
-    private auth: AngularFireAuth
-  ) { 
+  currentUser: any;
+   private authStatusSub = new BehaviorSubject<any>(null);
+   currentAuthStatus = this.authStatusSub.asObservable();
+
+  constructor(private auth: AngularFireAuth) { 
     this.authStatusListener();
   }
-  
-  currentUser: any ;
-  private authStatusSub = new BehaviorSubject(<any>null);
-  currentAuthStatus = this.authStatusSub.asObservable();
 
-  authStatusListener(){
-    this.auth.onAuthStateChanged((credential)=>{
-      if(credential){
-        console.log(credential);
-        this.authStatusSub.next(credential);
-        console.log('User is logged in');
-      }
-      else{
-        this.authStatusSub.next(null);
-        console.log('User is logged out');
-      }
-    })
+  authStatusListener() {
+    this.auth.authState.subscribe((credential) => {
+      this.currentUser = credential || null;
+      this.authStatusSub.next(this.currentUser);
+    });
   }
 
+  ggi() {
```

Comment:
en faire une fonction pour recuperer les données de l'utilisateur si connecté

ajouter un getter qui renvoit les données de l'utilisateur connecter via firebase un truc de ce genre : 
`getUser() : <fbUser>{
fbUser : fbUser;
this.currentAuthStatus.subscribe(user=>
    fbUser= user
    }
return fbuser;
}

`